text/0000-filesystem-integration.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: Filesystem Integration
+- Start Date: 2019-07-02
+- RFC PR:
+- Issue:
+
+# Summary
+[summary]: #summary
+
+Adds filesystem integration to qri datasets including a current working directory context for some qri commands.  The goal is to provide an experience that is more analogous to the git workflow, where working changes to files are sensed prior to committing a new version.
+
+# Motivation
+[motivation]: #motivation
+
+The motivation for this feature is to provide a more tangible connection between the qri versioning/publishing ecosystem and the file-based data that data practitioners are used to working with. Since all of the information in a qri dataset maps to files, we should just make those files exist automatically when a qri dataset is "added" to the user's local machine.  Likewise, we should watch for changes to those files, and ensure that they are all in a valid state before allowing the user to commit changes to a qri dataset.
+
+*This will significantly lower the barrier to entry for new users to qri, especially GUI users, as they can quickly pull down published datasets and begin working with them without having to understand how IPFS/QRI stores and maintains the data.*  
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The intent is to create a workflow more analogous with the git workflow.  When a new user wants the files in a remote git repo, they use `git clone {repoUrl}` and a new directory is created on their filesystem containing those files.  They can use _whatever tooling they want_ to make changes to the files, and commit them locally when they reach a critical point.
+
+We want the qri dataset user to have a similar experience: `qri add /b5/world_bank_population/` should create a new directory called `world_bank_population` on the filesystem, containing the following files:
+
+- `body.csv`
+- `meta.json`
+- `structure.json`
+- `transform.star`
+
+To change the dataset, the user uses _whatever tooling they want_ to modify the files, then commits the changes.
+
+## Commands
+
+Under this proposal, `qri use` is no longer relevant, as qri commands should be run from the qri dataset's directory.
+
+- `qri add` - creates a copy of the dataset's files in the present working directory
```

Comment:
`qri add` is an already existing command for fetching a dataset from the network. `qri add ramfox/nyc_candidates` will grab the dataset from either the network or. We could consider renaming the current `qri add` to `qri fetch` or `qri download`. Considering we're talking about mapping closer to git, we should consider such a change anyway, or at least make sure that users who try to use `add` in the same way as git don't create problems for themselves.

But there's a larger point to keep in mind here: Qri has no _staging area_, and I think that's a good thing. Because we have a declarative data model, we don't need to ask what to add. The user controls what's staged by manipulating the contents of the directory. Files that don't match our naming conventions aren't included. This means we can skip this "add" step entirely.